### ISSUE TITLE
Multiple input on a node using incorrect input for testing

### ISSUE THREAD

**Issue by pbrink231** [n8n-io/n8n#8556]
**Describe the bug**
This is a mix of a bug and a feature to fix the bug.  I have it labelled as a bug because of the unexpected behavior.  The Input being used for a step test is not clear and can be very misleading.  You receive a message of "Workflow executed successfully" but nothing is in the output.  On a module that has multiple nodes connected to its input, there is no way to know which input is being used to test the step.  This bug was actually very hard to pinpoint because the UI makes it look like everything is correct.  It also can cause what seems like strange behaviors because the input and step should do something but the output is showing something completely different.

**To Reproduce**
Create a flow that has 2 nodes point to the same node.   In that node run the "Test Step".  It is unclear which node is being used for the test.
[Test_Input_Bug.json](https://github.com/n8n-io/n8n/files/14172641/Test_Input_Bug.json)
In the example, you do not know if "Edit Fields1" or "Edit Fields2" is being used when running the test in "Edit Fields3" (sorry I didn't rename them).

![image](https://github.com/n8n-io/n8n/assets/8224560/9c50c27b-a7a6-40bc-ac4e-0f3ba7643f6e)

![image](https://github.com/n8n-io/n8n/assets/8224560/6a9c0040-0979-402b-a25f-f3c9c37c136a)

With pinning data which I used a lot in the situation because of the different possible flows you can wind up with really strange cases like the picture below.  And when the steps are more complicated it is very confusing.
json file for the photo below: 
[Test_Input_Bug (2).json](https://github.com/n8n-io/n8n/files/14172861/Test_Input_Bug.2.json)

![image](https://github.com/n8n-io/n8n/assets/8224560/9327e318-5a32-4a7e-a7d2-4ac29690d619)

![image](https://github.com/n8n-io/n8n/assets/8224560/9794631c-f583-4deb-864a-72f4b1cd1555)

**Expected behavior**
I was expecting the input fields on the left was the input used for the node.  I have come to find out that it is more of an explorer of past exports that does not have any affect on the current node.  I wrote a solution which is more of a feature request in the Additional context but something as simple as a popup when hovering over test step that gives the name of the input node being used.

![image](https://github.com/n8n-io/n8n/assets/8224560/a237a6f3-2432-47ae-aa1d-2024b65a3c59)

**Environment (please complete the following information):**

- OS: Ubuntu Linux 22 (Docker)
- n8n Version 1.25.1
- Node.js Version 18.16.0
- Database system SQLite
- Operation mode main

**Additional context**
This is where more of a feature request than a bug comes in but I would expect a label and hopefully a selection box showing which node is being used for the test.  Obviously using the Input on the left for selection would be better but this is atleast something to make it obvious which node is being tested against.

Something like this (I have no design sense and I am sure you could use a down arrow next to Test step for selection or something that actually looks good when more than 1 node is connected as an input):

![image](https://github.com/n8n-io/n8n/assets/8224560/34b54459-99b7-4751-ae92-69c08b5e0f37)

**Comment by Joffcom** [comment]
Hey @pbrink231,

I have taken a look at this and it looks you are seeing the same issue that was reported here: #7509 for the first part.

When the data is pinned it is the same the node will use the data from the path that is triggered from the switch node so in your screenshot the edit fields node is using the pinned data from the High branch as that is the route being taken to trigger the node this doesn't seem like a bug to me and is expected.

If both nodes were to trigger you would see the Run option in the output as the node would run once for each incoming input, This doesn't tell you which node ran to get the data in though so if you wanted to raise a feature request on the community support forum for this we can look at adding it.

![image](https://github.com/n8n-io/n8n/assets/4688521/2c120743-0f2a-4d72-81dd-538a5512fac2)

For now I believe as this is mostly a duplicate of the other issue I think we can close this, Let me know if I have missed anything or if you have any other concerns and we can open this again.

**Comment by pbrink231** [comment]
Thank you @Joffcom for taking a look.

I think you pointed out the issue exactly and I looked for previous issues but didn't find that one.

From your response I think I would request 2 features if they don't exist already and hopefully I can try and attack them myself.
- Output, Show which node name the input came from
- Test Step, drop down to choose which input to use
     - workflow (however the flow goes, currently used)
     - input 1
     - input 2
     - etc..

I will put them in later and update here.

**Comment by Joffcom** [comment]
Hey @pbrink231,

Based on your screenshot and what I spotted that appeared to be the case but it is also very possible that the same issue with first node only being used is in play as well.

**Comment by pbrink231** [comment]
@Joffcom 

I did a test.  I removed the "high side" and then added it again.  The dropdown input list showing past inputs now had "low side" first, then "high side".

When running the tests everything was reversed.  It now used "low side" as the input always instead of "high side".

I don't understand when the nodes are not pinned, why it runs the first time when the workflow is using the first input and then does not work at all after that.
